pkg/cmd/kafka/consumergroup/resetoffset/reset_offset.go

```diff
@@ -0,0 +1,209 @@
+
+	_ = cmd.MarkFlagRequired("id")
+	_ = cmd.MarkFlagRequired("offset")
+
+	// flag based completions for ID
+	_ = cmd.RegisterFlagCompletionFunc("id", func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		return cmdutil.FilterValidConsumerGroupIDs(f, toComplete)
+	})
+
+	// flag based completions for topic
+	_ = cmd.RegisterFlagCompletionFunc("topic", func(cmd *cobra.Command, _ []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		return cmdutil.FilterValidTopicNameArgs(f, toComplete)
+	})
+
+	flagutil.EnableStaticFlagCompletion(cmd, "offset", flagutil.ValidOffsets)
+
+	return cmd
+}
+
+// nolint:funlen
+func runCmd(opts *Options) error {
+
+	fmt.Println("reset offset")
+	logger, err := opts.Logger()
+	if err != nil {
+		return err
+	}
+
+	conn, err := opts.Connection(connection.DefaultConfigRequireMasAuth)
+	if err != nil {
+		return err
+	}
+
+	api, kafkaInstance, err := conn.API().KafkaAdmin(opts.kafkaID)
+	if err != nil {
+		return err
+	}
+
+	ctx := context.Background()
+
+	offsetResetParams := kafkainstanceclient.ConsumerGroupResetOffsetParameters{
+		Offset: opts.offset,
+	}
+
+	if opts.value != "" {
+		offsetResetParams.Value = &opts.value
+	}
+
+	if opts.topic != "" {
+		topicToReset := kafkainstanceclient.TopicsToResetOffset{
+			Topic: opts.topic,
+		}
+
+		if opts.partitions != "" {
+			partitionsStr := strings.Fields(opts.partitions)
+
+			partitionsArr := []int32{}
+
+			for _, partition := range partitionsStr {
+				partitionInt, convErr := strconv.ParseInt(partition, 10, 32)
+				if convErr != nil {
+					return convErr
+				}
+				partitionsArr = append(partitionsArr, int32(partitionInt))
+			}
+
+			topicToReset.Partitions = &partitionsArr
+		}
+
+		topicsToResetArr := []kafkainstanceclient.TopicsToResetOffset{topicToReset}
+
+		offsetResetParams.Topics = &topicsToResetArr
+	}
+
+	a := api.GroupsApi.ResetConsumerGroupOffset(ctx, opts.id).ConsumerGroupResetOffsetParameters(offsetResetParams)
+
+	if !opts.skipConfirm {
+
+		var confirmReset bool
+		opts.localizer.MustLocalize("kafka.consumerGroup.resetOffset.input.confirmReset.message", localize.NewEntry("ID", opts.id))
+		promptConfirmReset := &survey.Confirm{
+			Message: opts.localizer.MustLocalize("kafka.consumerGroup.resetOffset.input.confirmReset.message", localize.NewEntry("ID", opts.id)),
+		}
+
+		if err = survey.AskOne(promptConfirmReset, &confirmReset); err != nil {
+			return err
+		}
+		if !confirmReset {
+			logger.Debug(opts.localizer.MustLocalize("kafka.consumerGroup.resetOffset.log.debug.cancelledReset"))
+			return nil
+		}
+	}
+
+	_, httpRes, err := a.Execute()
+	fmt.Println("status code", httpRes.StatusCode)
+
+	if err != nil {
+
+		if httpRes == nil {
+			return err
+		}
+
+		operationTmplPair := localize.NewEntry("Operation", "reset offset")
+
+		switch httpRes.StatusCode {
+		case 401:
```

Comment:
Should ideally be `case http.StatusUnauthorized` etc.

Reply:
We should do it in a follow up PR for all the commands as well. We can create an issue for this.

Reply:
Agreed, yes that would be great.